### PR TITLE
Stripe 

### PR DESCRIPTION
**Pull request by BNMoyers**
# Description
   Integrates Stripe payment processing into the backend; creates the subscription route and links the Stripe customer information to the parent table.
Fixes # (issue)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Change Status

- [x] Complete, tested, ready to review and merge
- [ ] Complete, but not tested (may need new tests)
- [ ] Incomplete/work-in-progress, PR is for discussion/feedback

# How Has This Been Tested?

- [ ] Test A
- [ ] Test B

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My code has been reviewed by at least one peer
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] There are no merge conflicts

### CODE REVIEW

**Review by ghost — COMMENTED** [review]
token added

**Review by ghost — COMMENTED** [review]
rebasing of code.

**Review by ghost — COMMENTED** [review]
rebasing of code.

**Review by ghost — COMMENTED** [review]
link customer to parent table.

**Review by ghost — COMMENTED** [review]
rebasing of code

**Review by ghost — COMMENTED** [review]
rebasing of code and redoing of new customer route.

**Review by ghost — COMMENTED** [review]
reabsing of code for add-card.

**Review by ghost — COMMENTED** [review]
reabsing of code for add subscription

**Review by ghost — COMMENTED** [review]
changed code to handle error.

**Review by ghost — COMMENTED** [review]
rebasing of code.

**Review by ghost — COMMENTED** [review]
rebasing of code.

**Review by ghost — COMMENTED** [review]
clean up code.

**Comment by wSedlacek** [comment]
[[Parent] - Manage my children's account and preferences](https://trello.com/c/gLrPV5Wd/21-parent-manage-my-childrens-account-and-preferences)

**Comment by wSedlacek** [comment]
[[Parent] - Use a recognizable payment platform](https://trello.com/c/aOq56W7Y/23-parent-use-a-recognizable-payment-platform)

File: src/routes/canon/canon.routes.spec.ts [hunk 1]
```diff
@@ -40,11 +37,4 @@ describe('POST /canon', () => {
             .send({ week: 1, base64: 'pdf' })
             .expect(201);
     });
-
-    it('should return 400 with incorrect body', async () => {
-        await request(app)
-            .post('/canon')
-            .send({ base64: 'pdf', week: undefined })
-            .expect(400);
-    });
```
Comment (EricBanker12) [inline src/routes/canon/canon.routes.spec.ts:49]: Why was this test removed?

**Review by EricBanker12 — APPROVED** [review]
All tests pass, and I found everything working during manual testing.

The refactoring to other components unrelated to stripe implementation should have been done in a separate feature branch.

File: src/routes/canon/canon.routes.spec.ts (hunk 1 above)
Comment (wSedlacek) [reply src/routes/canon/canon.routes.spec.ts:49]: This is not a concern of the `canon.routes.ts` and should be moved to the validation middleware, this has not been done yet but will be done in the near future.

**Pull request merged**